xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/page/PageNotificationEventDescriptor.java

```diff
@@ -0,0 +1,174 @@
+
+    private String validationExpression;
+
+    private String notificationTemplate;
+
+    private DocumentReference authorReference;
+
+    private RecordableEventDescriptorContainer recordableEventDescriptorContainer;
+
+    /**
+     * Constructs a new PageNotificationEventDescriptor.
+     * We use a map in order to pass less arguments to the PageNotificationEventDescriptor constructor.
+     * See CheckStyle ParameterNumber rule.
+     *
+     * @param parameters A map of the objects parameters
+     * @param eventTriggerList A list of the events that should trigger the notification
+     * @param authorReference A reference to the XObject author (used in template rendering in order to avoid
+     * privilege escalation)
+     */
+    public PageNotificationEventDescriptor(
+            Map<String, String> parameters,
+            List<String> eventTriggerList,
+            DocumentReference authorReference)
+    {
+        this.applicationName = parameters.get("applicationName");
+        this.eventType = parameters.get("eventType");
+        this.eventPrettyName = parameters.get("eventPrettyName");
+        this.eventIcon = parameters.get("eventIcon");
+        this.eventTriggers = eventTriggerList;
+        this.objectType = parameters.get("objectType");
+        this.validationExpression = parameters.get("validationExpression");
+        this.notificationTemplate = parameters.get("notificationTemplate");
+        this.authorReference = authorReference;
+    }
+
+    /**
+     * Register the {@link PageNotificationEventDescriptor} to the {@link RecordableEventDescriptorContainer}.
+     *
+     * @param recordableEventDescriptorContainer Reference to the current {@link RecordableEventDescriptorContainer}
+     * component
+     */
+    public void register(RecordableEventDescriptorContainer recordableEventDescriptorContainer) {
+        this.recordableEventDescriptorContainer = recordableEventDescriptorContainer;
+        this.recordableEventDescriptorContainer.addRecordableEventDescriptor(this);
+    }
+
+    /**
+     * Unload the {@link PageNotificationEventDescriptor} from the {@link RecordableEventDescriptorContainer}.
+     */
+    public void unRegister() {
+        this.recordableEventDescriptorContainer.deleteRecordableEventDescriptor(this);
+    }
+
+    /**
+     * @return the custom event name
+     */
+    public String getEventName()
+    {
+        return this.eventType;
+    }
+
+    /**
+     * @return the object that should be listened
+     */
+    public String getObjectType()
+    {
+        return this.objectType;
+    }
+
+    /**
+     * @return the notification template
+     */
+    public String getNotificationTemplate()
+    {
+        return this.notificationTemplate;
+    }
+
+    /**
+     * @return the event canonical name
+     */
+    public List<String> getEventTriggers()
+    {
+        return this.eventTriggers;
+    }
+
+    /**
+     * @return the event validation expression
```

Comment:
This is an expression (in XWiki syntax) that, when evaluated, should return `true` for the notification to be sent.